cognite/neat/rules/models/_rules/dms_architect_rules.py

```diff
@@ -804,9 +825,17 @@ def _create_views_with_node_types(
         parent_views = {parent for view in views for parent in view.implements or []}
         for view in views:
             ref_containers = sorted(view.referenced_containers(), key=lambda c: c.as_tuple())
-            has_data = dm.filters.HasData(containers=list(ref_containers)) if ref_containers else None
-            node_type = dm.filters.Equals(["node", "type"], {"space": view.space, "externalId": view.external_id})
             dms_view = dms_view_by_id.get(view.as_id())
+            has_data = dm.filters.HasData(containers=list(ref_containers)) if ref_containers else None
+            if dms_view and isinstance(dms_view.reference, ReferenceEntity):
+                # If the view is a reference, we implement the reference view,
+                # and need the filter to match the reference
+                ref_view = dms_view.reference.as_id(False, default_space, default_version, self.standardize_casing)
```

Comment:
aren't we able to pick up space and version from reference ? would this lead to different space/version?

Reply:
I don't think this can happen in practice, but the method requires it. There is some technical debt around the entity classes.